File: 0242-valid-anagram/solution.c
bool isAnagram(char* s, char* t) {
    int n=strlen(t);
    int m=strlen(s);
    if(n!=m)return false;
    int arr[126]={0};
    for(int i=0;i<n;i++){
        arr[s[i]]++;
        arr[t[i]]--;
    }
    for(int i=0;i<n;i++){
        if(arr[s[i]]!=0){
            return false;
        }
    }
    return true;
}

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer> map= new HashMap<>();
        int n=s.length();
        int m=t.length();
        if(n!=m)return false;
        for(char a:s.toCharArray()){
            map.put(a,map.getOrDefault(a,0)+1);
        }
        for(char a:t.toCharArray()){
            map.put(a,map.getOrDefault(a,0)-1);
        }
        for(char a:s.toCharArray()){
            if(map.get(a)!=0){
                return false;
            }
        }
        return true;
    }
}
